subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -62,11 +430,62 @@ uint8_t ll_big_test_create(uint8_t big_handle, uint8_t adv_handle,
 	return BT_HCI_ERR_CMD_DISALLOWED;
 }
 
-uint8_t ll_big_terminate(uint8_t big_handle, uint8_t reason)
+
+static void tx_lll_flush(void *param)
 {
-	/* TODO: Implement */
-	ARG_UNUSED(big_handle);
-	ARG_UNUSED(reason);
+	/* TODO: LLL support for ADV ISO */
+#if 0
+	/* TODO: Flush TX */
+	struct ll_adv_iso *lll = param;
+#endif
+}
 
-	return BT_HCI_ERR_CMD_DISALLOWED;
+static void ticker_op_stop_cb(uint32_t status, void *param)
+{
+	uint32_t retval;
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, tx_lll_flush};
+
+	LL_ASSERT(status == TICKER_STATUS_SUCCESS);
+
+	mfy.param = param;
+
+	/* Flush pending tx PDUs in LLL (using a mayfly) */
+	retval = mayfly_enqueue(TICKER_USER_ID_ULL_LOW, TICKER_USER_ID_LLL, 0,
+				&mfy);
+	LL_ASSERT(!retval);
+}
+
+uint8_t ll_big_terminate(uint8_t big_handle, uint8_t reason, void **rx)
+{
+	struct ll_adv_iso *adv_iso;
+	struct node_rx_pdu *node_rx;
+	uint32_t ret;
+
+	adv_iso = ull_adv_iso_get(big_handle);
+
+	if (!adv_iso) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	/* TODO: Terminate all BIS data paths */
+
+	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_HIGH,
+			  TICKER_ID_ADV_ISO_BASE + adv_iso->bis_handle,
+			  ticker_op_stop_cb, adv_iso);
+
+	adv_iso->node_rx_terminate.reason = reason;
```

Comment:
I think reason could be assigned directly to `node_rx->pdu` below instead of going through `node_rx_terminate_reason`. The `reason` field should alias `node_rx->pdu` anyway.

Reply:
Sure. Personally I think assigning values to unnamed values (e.g. `node_rx->pdu`) is less readable than assigning the value to e.g. `adv_iso->node_rx_terminate.reason`, but it will be more consistent to other uses.